chainer/distribution.py

```diff
@@ -0,0 +1,420 @@
+            evaluated at `x`.
+
+        """
+        raise NotImplementedError
+
+    def log_cdf(self, x):
+        """Evaluates the log of cumulative distribution function at a given points.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Data points in the domain of the
+            distribution
+
+        Returns:
+            ~chainer.Variable: Logarithm of cumulative distribution function
+            value evaluated at `x`.
+
+        """
+        raise NotImplementedError
+
+    def log_prob(self, x):
+        """Evaluates the logarithm of probability at a given points.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Data points in the domain of the
+            distribution
+
+        Returns:
+            ~chainer.Variable: Logarithm of probability evaluated at `x`.
+
+        """
+        raise NotImplementedError
+
+    def log_survival_function(self, x):
+        """Evaluates the logarithm of survival function at a given points.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Data points in the domain of the
+            distribution
+
+        Returns:
+            ~chainer.Variable: Logarithm of survival function value evaluated
+            at `x`.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def mean(self):
+        """Returns the mean of the distribution.
+
+        Returns:
+            ~chainer.Variable: The mean of the distribution.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def mode(self):
+        """Returns the mode of the distribution.
+
+        Returns:
+            ~chainer.Variable: The mode of the distribution.
+
+        """
+        raise NotImplementedError
+
+    def perplexity(self, x):
+        """Evaluates the perplexity function at a given points.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Data points in the domain of the
+            distribution
+
+        Returns:
+            ~chainer.Variable: Perplexity function value evaluated at `x`.
+
+        """
+        raise NotImplementedError
+
+    def prob(self, x):
+        """Evaluates probability at a given points.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Data points in the domain of the
+            distribution
+
+        Returns:
+            ~chainer.Variable: Probability evaluated at `x`.
+
+        """
+        raise NotImplementedError
+
+    def sample(self, sample_shape=()):
+        """Samples random points from the distribution.
+
+        This function call `sample_n` and reshape a result of `sample_n` to
+        `(sample_shape, batch_shape, event_shape)`. On implementing sampling
```

Comment:
Better to write `sample_shape + batch_shape + event_shape` rather than `(sample_shape, batch_shape, event_shape)`.